listenbrainz/webserver/static/js/src/LastFMImporter.tsx

```diff
@@ -151,6 +155,20 @@ export default class LastFmImporter extends React.Component<
     }
   }
 
+  async getUserPrivacy() {
+    // fetch page from Last.fm
+    const { lastfmUsername } = this.state;
+    const url = `${this.lastfmURL}?method=user.getrecenttracks&user=${lastfmUsername}&api_key=${this.lastfmKey}&format=json`;
+    const response = await fetch(encodeURI(url));
```

Comment:
Let's add a try-catch block around this await to catch any errors with that fetch call.

What should happen if the fetch fails ? (for example a temporary network issue)
Once that is decided, you can add a test with a simulated failure of the fetch with something like `window.fetch.mockImplementationOnce(() => Promise.reject("Some error"));`